opal/mca/btl/ofi/btl_ofi_component.c

```diff
@@ -46,6 +46,16 @@
 
 #define MCA_BTL_OFI_REQUESTED_MR_MODE (FI_MR_ALLOCATED | FI_MR_PROV_KEY | FI_MR_VIRT_ADDR | FI_MR_ENDPOINT)
 
+/**
+ * FI_MR_BASIC (1<<0) and FI_MR_SCALABLE (1<<1) are deprecated
+ * since Libfabric 1.5 and the symbols will get removed in
+ * future Libfabric 2.x versions. Use the internal mode bits
+ * for backward compatibilities without breaking compilation
+ * with newer libfabric.
```

Comment:
This assumes that a newer libfabric won't be reusing those old values?  Otherwise FI_MR_SUPER_NEW_FANCY_MODE with value (1<<0) might be misinterpreted as FI_MR_BASIC in some future compile.

Reply:
I would hope ofi wg  would not make that mistake.

Reply:
Yes, there won't be newer libfabric to use these bits, it's forbidden in the development

Reply:
okay I think in this case we will merge this PR.  Thanks for cleaning our code a bit @shijin-aws